pkg/controller/performanceprofile/components/tuned/tuned.go

```diff
@@ -126,18 +126,23 @@ func NewNodePerformance(assetsDir string, profile *performancev2.PerformanceProf
 
 		for _, device := range profile.Spec.Net.Devices {
 			devices = make([]string, 0)
-			if device.InterfaceName != nil {
-				devices = append(devices, *device.InterfaceName)
+			if device.DeviceID != nil {
+				devices = append(devices, "^ID_MODEL_ID="+*device.DeviceID)
 			}
 			if device.VendorID != nil {
-				devices = append(devices, *device.VendorID)
+				devices = append(devices, "^ID_VENDOR_ID="+*device.VendorID)
 			}
-			if device.DeviceID != nil {
-				devices = append(devices, *device.DeviceID)
+			if device.InterfaceName != nil {
+				deviceNameAmendedRegex := strings.Replace(*device.InterfaceName, "*", ".*", -1)
+				if strings.HasPrefix(*device.InterfaceName, "!") {
+					devices = append(devices, "^INTERFACE_NAME="+"(?!"+deviceNameAmendedRegex+")")
+				} else {
+					devices = append(devices, "^INTERFACE_NAME="+deviceNameAmendedRegex)
+				}
 			}
-			devicesArgs := strings.Join(devices, ",")
+			devicesUdevRegex := "r'" + strings.Join(devices, `[\s\S]*`) + "'"
```

Comment:
could you please add a comment about how the final regex should look like? regexes may be hard enough, having them built programmatically is even harder.